src/sentry/identity/vsts/provider.py

```diff
@@ -135,3 +135,108 @@ def exchange_token(self, request: Request, pipeline, code):
         if req.headers["Content-Type"].startswith("application/x-www-form-urlencoded"):
             return dict(parse_qsl(body))
         return orjson.loads(body)
+
+
+class VSTSNewIdentityProvider(OAuth2Provider):
```

Comment:
Please add some comments on what this is. I can imagine it'll be very confusing to future engineers if we end up not cleaning up fast